xbmc/linux/CMakeLists.txt

```diff
@@ -1,3 +1,6 @@
+file(GLOB VARIANTS_CPP ${CORE_SYSTEM_VARIANT}/*.cpp)
+file(GLOB VARIANTS_H ${CORE_SYSTEM_VARIANT}/*.h)
```

Comment:
There's one problem with globbing of source files: When you add/remove a file but don't touch any CMake files, then CMake doesn't know that it needs to rerun and you'll get some difficult to understand compile/link errors. Also if I'm not mistaken, CORE_SYSTEM_VARIANT is currently not userd/set for CMake.

@wsnipex, @notspiff: Do you already have a solution for those variants in mind? It should probably be an own lib that gets added through the treedata structure, right? Then we have to decide if we want to handle it like rbpi (with it's own platform config in project/cmake/platform) or we add variant support to linux.